packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -308,18 +308,7 @@ class OverlayState extends State<Overlay> with TickerProviderStateMixin {
   ///
   /// It is an error to specify both `above` and `below`.
   void insert(OverlayEntry entry, { OverlayEntry below, OverlayEntry above }) {
-    assert(
-      above == null || below == null,
-      'Only one of `above` and `below` may be specified.',
-    );
-    assert(
-      above == null || (above._overlay == this && _entries.contains(above)),
-      'The provided entry for `above` is not present in the Overlay.',
-    );
-    assert(
-      below == null || (below._overlay == this && _entries.contains(below)),
-      'The provided entry for `below` is not present in the Overlay.',
-    );
+    _assertEntriesInsertedPosition(above, below);
```

Comment:
Can you wrap this call in an assert so it gets optimized out in release builds?

Also, the method name should start with `_debug`, maybe _debugVerifyInsertPosition.